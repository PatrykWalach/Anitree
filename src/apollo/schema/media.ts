import { ScoreFormat } from './viewer'
import { MediaList } from './mediaListCollection'

export interface Variables {
  isAdult?: boolean
  id: number
  format?: ScoreFormat
}

export interface Node {
  id: number
  type: MediaType
}

export type MediaType = 'MANGA' | 'ANIME'
export interface FuzzyDate {
  year: number | null
  month: number | null
  day: number | null
}

export interface Media extends Node {
  startDate: FuzzyDate
  endDate: FuzzyDate
  format: string | null
  description: string | null
  season: string | null
  seasonInt: number | null
  episodes: number | null
  chapters: number | null
  volumes: number | null
  isFavourite: boolean
  mediaListEntry: MediaList | null
  title: MediaTitle
  siteUrl: string
  relations: {
    edges: MediaEdge[]
  }
  tags: {
    rank: number
    name: string
  }[]
  studios: {
    nodes: Studio[]
  }
  status: MediaStatus | null
  bannerImage: string | null
  coverImage: MediaCoverImage
  stats: MediaStats
}
export interface MediaStats {
  scoreDistribution: ScoreDistribution[] | null
}
export interface ScoreDistribution {
  score: number
  amount: number
}
export interface MediaTitle {
  romaji: string
  english: string
  native: string
}

export type MediaRelation =
  | 'ADAPTATION'
  | 'PREQUEL'
  | 'SEQUEL'
  | 'PARENT'
  | 'SIDE_STORY'
  | 'CHARACTER'
  | 'SUMMARY'
  | 'ALTERNATIVE'
  | 'SPIN_OFF'
  | 'OTHER'
  | 'SOURCE'
  | 'COMPILATION'
  | 'CONTAINS'

export interface MediaEdge {
  node: Node
  relationType: MediaRelation
  id: number
}

export interface Studio {
  id: number
  name: string
}

export interface MediaCoverImage {
  extraLarge: string | null
  large: string | null
  medium: string | null
  color: string | null
}

export type MediaStatus =
  | 'CURRENT'
  | 'PLANNING'
  | 'COMPLETED'
  | 'DROPPED'
  | 'PAUSED'
  | 'REPEATING'
