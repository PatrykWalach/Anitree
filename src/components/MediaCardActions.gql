#import './MediaEdit.gql'
#import '../hooks/ToggleFavourite.gql'
#import '../hooks/Share.gql'

fragment MediaCardActions_media on Media {
  id
  type
  siteUrl
  ...ToggleFavourite_media
  ...Share_media
  ...MediaEdit_media
}


