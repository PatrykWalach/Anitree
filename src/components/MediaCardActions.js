import { MediaEdit_media } from './MediaEdit.js'
import { ToggleFavourite_media } from '../hooks/ToggleFavourite.js'
import { Share_media } from '../hooks/Share.js'

import gql from 'graphql-tag'

export const MediaCardActions_media = gql`
  fragment MediaCardActions_media on Media {
    id
    type
    siteUrl
    ...ToggleFavourite_media
    ...Share_media
    ...MediaEdit_media
  }
  ${ToggleFavourite_media}
  ${Share_media}
  ${MediaEdit_media}
`
