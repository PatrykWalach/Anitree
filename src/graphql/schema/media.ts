import { ScoreFormat } from './viewer'
import { MediaList } from './mediaListCollection'

export interface Variables {
  isAdult?: boolean
  id: number
  format?: ScoreFormat
}

export interface Node {
  __typename: 'Media'
  id: number
  type: MediaType
}

export type MediaType = 'MANGA' | 'ANIME'
export interface FuzzyDate {
  __typename: 'FuzzyDate'
  year: number | null
  month: number | null
  day: number | null
}

export interface Media extends Node {
  startDate: FuzzyDate
  endDate: FuzzyDate
  format: string | null
  description: string | null
  season: string | null
  seasonInt: number | null
  episodes: number | null
  chapters: number | null
  volumes: number | null
  isFavourite: boolean
  ranknigs: MediaRank[]
  mediaListEntry: MediaList | null
  title: MediaTitle
  siteUrl: string
  relations: MediaConnection
  tags: MediaTag[]
  studios: StudioConnection
  status: MediaStatus | null
  bannerImage: string | null
  coverImage: MediaCoverImage
  stats: MediaStats
}
export interface MediaConnection {
  __typename: 'MediaConnection'
  edges: MediaEdge[]
}
export interface MediaTag {
  __typename: 'MediaTag'
  id: number
  rank: number
  name: string
}
export interface StudioConnection {
  __typename: 'StudioConnection'

  nodes: Studio[]
}
export interface MediaRank {
  __typename: 'MediaRank'
  allTime: boolean
  context: string
  rank: number
  type: 'RATED' | 'POPULAR'
  year: null | number
}
export interface MediaStats {
  __typename: 'MediaStats'
  scoreDistribution: ScoreDistribution[] | null
}
export interface ScoreDistribution {
  __typename: 'ScoreDistribution'
  score: number
  amount: number
}
export interface MediaTitle {
  __typename: 'MediaTitle'
  romaji: string
  english: string
  native: string
}

export type MediaRelation =
  | 'ADAPTATION'
  | 'PREQUEL'
  | 'SEQUEL'
  | 'PARENT'
  | 'SIDE_STORY'
  | 'CHARACTER'
  | 'SUMMARY'
  | 'ALTERNATIVE'
  | 'SPIN_OFF'
  | 'OTHER'
  | 'SOURCE'
  | 'COMPILATION'
  | 'CONTAINS'

export interface MediaEdge {
  __typename: 'MediaEdge'
  node: Node
  relationType: MediaRelation
  id: number
}

export interface Studio {
  __typename: 'Studio'
  id: number
  name: string
}

export interface MediaCoverImage {
  __typename: 'MediaCoverImage'
  extraLarge: string | null
  large: string | null
  medium: string | null
  color: string | null
}

export type MediaStatus =
  | 'CURRENT'
  | 'PLANNING'
  | 'COMPLETED'
  | 'DROPPED'
  | 'PAUSED'
  | 'REPEATING'
