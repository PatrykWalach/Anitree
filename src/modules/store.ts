import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
