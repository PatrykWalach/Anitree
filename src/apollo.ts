import { ApolloLink, concat } from 'apollo-link'
import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory'
import { PersistedData, PersistentStorage } from 'apollo-cache-persist/types'

import { ApolloClient } from 'apollo-client'
import { HttpLink } from 'apollo-link-http'

import { Variables } from '@/graphql/schema/media'
import Vue from 'vue'
import VueApollo from 'vue-apollo'
import { persistCache } from 'apollo-cache-persist'

import { store } from '@/store'

const link = new HttpLink({
  uri: 'https://graphql.anilist.co',
})

const middle = new ApolloLink((operation, forward) => {
  const token = store.getState().settings.token

  if (token) {
    operation.setContext({
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
  }
  if (forward) {
    return forward(operation)
  }
  return null
})

const cache = new InMemoryCache({
  cacheRedirects: {
    Query: {
      Media: (_, args, { getCacheKey }) =>
        getCacheKey({ __typename: 'Media', id: (args as Variables).id }),
    },
  },
})

// if (store.getState()settings..cacheApollo) {
persistCache({
  cache,
  storage: window.localStorage as PersistentStorage<
    PersistedData<NormalizedCacheObject>
  >,
})
// }

export const apollo = new ApolloClient({
  cache,
  link: concat(middle, link),
})

Vue.use(VueApollo)

export const apolloProvider = new VueApollo({
  defaultClient: apollo,
})
