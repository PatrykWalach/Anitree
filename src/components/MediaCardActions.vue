<template>
  <v-card-actions>
    <v-btn
      text
      color="primary"
      :disabled="!media"
      :to="{
        name: 'media',
        params: {
          mediaId: media.id,
          mediaType: media.type.toLowerCase(),
        },
      }"
    >
      Explore
    </v-btn>

    <v-spacer></v-spacer>

    <v-tooltip top>
      <template v-slot:activator="{ attrs, on }">
        <v-btn
          v-bind="attrs"
          icon
          :disabled="!media"
          rel="noopener"
          target="_blank"
          :href="url"
          v-on="on"
        >
          <v-icon>open_in_new</v-icon>
        </v-btn>
      </template>
      <span>Anilist</span>
    </v-tooltip>

    <v-tooltip top>
      <template v-slot:activator="{ attrs, on }">
        <v-btn
          v-bind="attrs"
          icon
          :disabled="!media"
          v-on="on"
          @click.stop="share"
        >
          <v-icon>share</v-icon>
        </v-btn>
      </template>
      <span>Share</span>
    </v-tooltip>

    <v-tooltip v-if="Viewer" top>
      <template v-slot:activator="{ attrs, on }">
        <v-btn
          v-bind="attrs"
          icon
          :disabled="!media"
          v-on="on"
          @click.stop="edit"
        >
          <v-icon>edit</v-icon>
        </v-btn>
      </template>
      <span>Edit</span>
    </v-tooltip>
  </v-card-actions>
</template>

<script lang="ts">
import { NewNavigator, ShareData } from '../types'
import { computed, createComponent } from '@vue/composition-api'

import { Media } from '@/graphql/schema/media'
import { VIEWER } from '@/graphql'

import { edit as editModule } from '../store/edit'
import { settings } from '../store/settings'
import { share as shareModule } from '../store/share'
import { title as titleModule } from '../store/title'

function useShare(props: Props) {
  const {
    mutations: { CHANGE_OPTIONS, CHANGE_IS_SHARED },
  } = shareModule

  const {
    getters: { getTitle },
  } = titleModule

  const desktopShare = async (data: ShareData) => {
    CHANGE_OPTIONS(data)
    CHANGE_IS_SHARED(true)
  }

  const title = computed(() => getTitle.value(props.media && props.media.title))

  const url = computed(() => (props.media && props.media.siteUrl) || '')

  const share = () => {
    const share = (navigator as NewNavigator).share || desktopShare

    share.call(navigator, {
      title: title.value,
      url: url.value,
    })
  }

  return { share, title, url }
}

export interface Props {
  media: Media | null
}

export default createComponent<Readonly<Props>>({
  apollo: {
    Viewer: {
      query: VIEWER,
      skip() {
        return !settings.state.token.value
      },
    },
  },
  props: {
    media: { default: null, required: true, type: null },
  },
  setup(props) {
    const { media } = props

    const {
      actions: { open },
    } = editModule

    const edit = () => {
      if (media) {
        open(media.id)
      }
    }

    return {
      edit,
      ...useShare(props),
    }
  },
})
</script>
