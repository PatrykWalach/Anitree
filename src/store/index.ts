import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
