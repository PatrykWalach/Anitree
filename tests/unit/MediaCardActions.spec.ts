import MediaCardActions from '@/components/MediaCardActions.vue'

import { matchSnapshot } from './utils'
import { mockedMedia } from './mocks/media'
import { useMockedApollo } from './mocks/apollo'
import { useMockedStore } from './mocks/store'

describe('MediaCardActions', () => {
  matchSnapshot(MediaCardActions, {
    propsData: {
      media: mockedMedia,
    },
    provide: {
      ...useMockedStore(),
      ...useMockedApollo(),
    },
  })
})
