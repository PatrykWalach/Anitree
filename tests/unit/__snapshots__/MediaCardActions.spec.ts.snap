// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCardActions matches snapshot 1`] = `
<v-card-actions-stub>
  <v-btn-stub
    activeclass=""
    color="primary"
    exact="true"
    tag="button"
    text="true"
    to="[object Object]"
    type="button"
  >
    
    Explore
  
  </v-btn-stub>
   
  <v-spacer-stub />
   
  <v-tooltip-stub
    closedelay="0"
    contentclass=""
    fixed="true"
    maxwidth="auto"
    nudgebottom="0"
    nudgeleft="0"
    nudgeright="0"
    nudgetop="0"
    nudgewidth="0"
    opendelay="0"
    openonhover="true"
    tag="span"
    top="true"
  >
     
    <span>
      Edit
    </span>
  </v-tooltip-stub>
  <v-tooltip-stub
    closedelay="0"
    contentclass=""
    fixed="true"
    maxwidth="auto"
    nudgebottom="0"
    nudgeleft="0"
    nudgeright="0"
    nudgetop="0"
    nudgewidth="0"
    opendelay="0"
    openonhover="true"
    tag="span"
    top="true"
  >
     
    <span>
      Share
    </span>
  </v-tooltip-stub>
  <v-tooltip-stub
    closedelay="0"
    contentclass=""
    fixed="true"
    maxwidth="auto"
    nudgebottom="0"
    nudgeleft="0"
    nudgeright="0"
    nudgetop="0"
    nudgewidth="0"
    opendelay="0"
    openonhover="true"
    tag="span"
    top="true"
  >
     
    <span>
      Anilist
    </span>
  </v-tooltip-stub>
</v-card-actions-stub>
`;
